main.py

```
@@ -1,603 +1,1133 @@
+
+class MarkdownScraper:
+    """Scrapes websites and converts content to markdown, JSON, or XML with chunking support."""
+
+    def __init__(
+        self,
+        requests_per_second: float = 1.0,
+        timeout: int = 30,
+        max_retries: int = 3,
+        chunk_size: int = 1000,
+        chunk_overlap: int = 200,
+        cache_enabled: bool = True,
```

Ruff (0.8.2)

Boolean-typed positional argument in function definition

(FBT001)